lib/rubocop/cop/rails/default_scope.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop looks for using of `default_scope`.
+      #
+      # @example
+      #   # bad
+      #   default_scope -> { where(hidden: false) }
+      #
+      #   # good
+      #   scope :published, -> { where(hidden: false) }
+      #
+      #   # bad
+      #   def self.default_scope
+      #     where(hidden: false)
+      #   end
+      #
+      #   # good
+      #   def self.published
+      #     where(hidden: false)
+      #   end
+      #
+      class DefaultScope < Cop
+        MSG = 'Avoid using `default_scope`. It is better to use explicitly named scopes.'
```

Comment:
I wonder if this can be phrased to indicate that using a scope is not the only alternative. For example, hidden records could be stored in a separate table.

Reply:
I think, most of the time the solution is just to name a scope. I'm fine with message, but maybe it is better to just remove the second sentence.